specification/cognitiveservices/data-plane/CustomDecisionService/v2.0/CustomDecisionService.json

```diff
@@ -0,0 +1,358 @@
+﻿{
+  "x-generator": "NSwag v11.12.7.0 (NJsonSchema v9.10.6.0 (Newtonsoft.Json v9.0.0.0))",
+  "swagger": "2.0",
+  "info": {
+    "title": "",
+    "version": ""
+  },
+  "schemes": [],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/api/v2/{appId}/rank": {
+      "post": {
+        "tags": [
+          "RankingServerFeatures"
+        ],
+        "operationId": "Rank",
+        "parameters": [
+          {
+            "type": "string",
```

Comment:
You have a lot of identical parameter definitions. For smaller Swagger (less duplication) *and* more control over generated code I recommend the following:

Move out any common parameter definitions into a dedicated `parameters` section and only `$ref` those parameters here. You can then use `x-ms-parameter-location: method/client` to control, whether you would like those parameters to appear on the client (as a property; makes sense if you'd expect this to stay the same across many calls, e.g. `appId`) or on the generated methods (so programmers have to pass them with every call).

Here is a complete example of all that: https://github.com/Azure/azure-rest-api-specs/blob/ab9b07c885234faf8a424f7868f95365fcfb9af2/specification/sql/resource-manager/Microsoft.Sql/2015-05-01/usages.json#L214